### ISSUE TITLE
Documentation unclear of how the memoizer is used

### ISSUE THREAD

**Issue by Mr-Wallet** [reduxjs/reselect#455]
The README describes how createSelectorCreator can be used to override the defaultMemoizer and how it will be called with the resultFunc and additional memoizeOptions. However, if my reading of the source code is correct, it's _also_ called before calling the input functions to short-circuit _their_ evaluation, and that _isn't_ called with the memoizeOptions. These are important details for implementing a custom memoizer, so it seems like an oversight in the documentation.

**Comment by srubin** [comment]
This bit me recently. I'd appreciate being able to specify a different memoize function for the two purposes. I guess I could do this using `memoizeOptions`:

```ts
function memoizeInputArgs(fn) { /* ... */ }
function memoizeResultArgs(fn) { /* ... */ }

const selector = createSelectorCreator((fn, isResultFunction) =>
  isResultFunction ? memoizeResultArgs(fn) : memoizeInputArgs(fn), true)
);
```

but it might be cleaner to allow passing a second memoize function to `createSelectorCreator`.

**Comment by srubin** [comment]
I ended up with this slightly more fleshed-out version:

```ts
export function myCreateSelectorCreator<F extends Function, T, R>(
    memoizeResultFnArgs: MemoizeFn,
    memoizeResultFnArgsOptions: any[] = [],
    memoizeInputFnArgs: MemoizeFn = defaultMemoize,
    memoizeInputFnArgsOptions: any[] = [],
) {
    function memoize(fn: Function, areResultArgs: boolean) {
        if (areResultArgs) {
            return memoizeResultFnArgs(fn, ...memoizeResultFnArgsOptions);
        }
        return memoizeInputFnArgs(fn, ...memoizeInputFnArgsOptions);
    }
    return createSelectorCreator(
        memoize,
        true, // only passed to `memoize` for result args, not input args
    );
}
```

**Comment by aryaemami59** [comment]
This one has been resolved by #636.
